service/src/main/java/com/derpgroup/derpwizard/resource/AlexaResource.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (C) 2015 David Phillips
+ * Copyright (C) 2015 Eric Olson
+ * Copyright (C) 2015 Rusty Gerard
+ * Copyright (C) 2015 Paul Winters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.derpgroup.derpwizard.resource;
+
+import io.dropwizard.setup.Environment;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.amazon.speech.json.SpeechletRequestEnvelope;
+import com.amazon.speech.json.SpeechletResponseEnvelope;
+import com.amazon.speech.speechlet.SpeechletRequest;
+import com.amazon.speech.speechlet.SpeechletResponse;
+import com.amazon.speech.ui.SimpleCard;
+import com.amazon.speech.ui.SsmlOutputSpeech;
+import com.derpgroup.derpwizard.configuration.MainConfig;
+import com.derpgroup.derpwizard.voice.alexa.AlexaRequestType;
+import com.derpgroup.derpwizard.voice.alexa.AlexaSkillsKitUtil;
+
+/**
+ * REST APIs for requests generating from Amazon Alexa
+ *
+ * @author Eric
+ * @since 0.0.1
+ */
+@Path("/alexa")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class AlexaResource {
+
+  public AlexaResource(MainConfig config, Environment env) {}
+
+  /**
+   * Generates a welcome message.
+   *
+   * @return The message, never null
+   */
+  @POST
+  public SpeechletResponseEnvelope doAlexaRequest(SpeechletRequestEnvelope request){
+    if(request == null || request.getRequest() == null){
```

Comment:
So, that's neat, and I'll make the change for now, but here's something we're going to need to figure out - right now dropwizard has a default exception mapper for the validation it does.  Unfortunately, the format it serializes into is not readable by Alexa.  We can change the exception mapper to return an Alexa formatted exception, but from what I can tell the mapper is defined in configuration and applies to the entire dropwizard instance.  What that means is that there's not a per-endpoint way to have the annotations map errors to different formats.  Bad news once we have non-Alexa resources.

We'll need to figure out a solution at that time.